common/lib/dependabot/notices.rb

```diff
@@ -0,0 +1,139 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+module Dependabot
+  class Notice
+    extend T::Sig
+
+    sig { returns(String) }
+    attr_reader :mode, :type
+
+    sig { returns(T.nilable(String)) }
+    attr_reader :package_manager_name, :message, :markdown
+
+    sig { returns(T::Hash[Symbol, T.untyped]) }
+    attr_reader :details
+
+    sig do
+      params(
+        mode: String,
+        type: String,
+        package_manager_name: T.nilable(String),
+        message: T.nilable(String),
```

Comment:
can we make this String instead of nilable String? I'm assuming we can then check for empty strings and eliminate the chance of nilerrors